crates/ibc/src/applications/transfer/relay/send_transfer.rs

```diff
@@ -1,33 +1,109 @@
-use crate::applications::transfer::context::TokenTransferContext;
+use crate::applications::transfer::context::{
+    TokenTransferExecutionContext, TokenTransferValidationContext,
+};
 use crate::applications::transfer::error::TokenTransferError;
 use crate::applications::transfer::events::TransferEvent;
 use crate::applications::transfer::msgs::transfer::MsgTransfer;
 use crate::applications::transfer::packet::PacketData;
 use crate::applications::transfer::{is_sender_chain_source, Coin, PrefixedCoin};
-use crate::core::ics04_channel::handler::send_packet::send_packet;
+use crate::core::ics04_channel::handler::send_packet::{send_packet_execute, send_packet_validate};
 use crate::core::ics04_channel::packet::Packet;
 use crate::core::ics24_host::path::{ChannelEndPath, SeqSendPath};
 use crate::events::ModuleEvent;
-use crate::handler::{HandlerOutput, HandlerOutputBuilder};
 use crate::prelude::*;
 
-/// This function handles the transfer sending logic.
-/// If this method returns an error, the runtime is expected to rollback all state modifications to
-/// the `Ctx` caused by all messages from the transaction that this `msg` is a part of.
-pub fn send_transfer<Ctx, C>(
-    ctx: &mut Ctx,
-    output: &mut HandlerOutputBuilder<()>,
+pub fn send_transfer<C, Ctx>(ctx_a: &mut Ctx, msg: MsgTransfer<C>) -> Result<(), TokenTransferError>
+where
+    C: TryInto<PrefixedCoin> + Clone,
+    Ctx: TokenTransferExecutionContext,
+{
+    send_transfer_validate(ctx_a, msg.clone())?;
+    send_transfer_execute(ctx_a, msg)
+}
+
+pub fn send_transfer_validate<C, Ctx>(
+    ctx_a: &Ctx,
```

Comment:
A side opinion about our naming convention (my experience so far)

While the convention for message fields has been super helpful, I think we can ease up on naming contexts. (in particualr convection contains `*_on_a` and `*_of_a` , not `*_a`) We only implement one context for each host and always interact with that one context. Keeping `ctx` instead of `ctx_a` and `ctx_b` would be just as simple, effective, without adding unnecessary complexity or suffixes. Yet, wherever there could be a confusion, we still see our naming convention around on a message related field or variable.

Reply:
Sorry, I'm not sure I understand. Are you suggesting we use `ctx` her here instead of `ctx_a`?

Reply:
Right. in general, I meant. Not just here

Reply:
This is following our convention (specifically [this part](https://github.com/cosmos/ibc-rs/blob/2f1fce19a5a14214682a09d5d19acf31470e4b48/crates/ibc/src/core/mod.rs#L16-L18)). I like this use of the convention, for example [here](https://github.com/cosmos/ibc-rs/blob/2f1fce19a5a14214682a09d5d19acf31470e4b48/crates/ibc/src/applications/transfer/relay/send_transfer.rs#L39), having `ctx_a` makes it easy to know I need to select `port_on_a` and `chan_on_a` to get the `ChannelEndPath` on "this chain" (the sending chain).

WDYT?

Reply:
Let's go with the present (ctx_a) format 👍🏻